File: old version/CardGame/CardGame/Base/Player.cs
using CardGame.Models;

namespace CardGame.Base
{
    public abstract class Player
    {
        public string Name { get; set; }

        public int Point { get; set; }

        public Hand Hand { get; set; } = new Hand();

        public void Namehimself(string name)
        {
            Name = name;
        }

        public abstract Card Play();
    }
}

File: C2/C2M2/c2m2h1-card-game-framework/game/base/Player.java
package game.base;


/**
 * @author - johnny850807@gmail.com (Waterball)
 */
public abstract class Player<Card> {
    protected Hand<Card> hand = new Hand<>();
    protected String name;

    public abstract void nameSelf(int order);

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void addHandCard(Card card) {
        this.hand.addCard(card);
    }

    public Hand<Card> getHand() {
        return hand;
    }
}
